### PR TITLE
IOS-17850: fix deleting index+extensionBgIDs prematurely

### PR DESCRIPTION
**Pull request by thuyBox**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @thuyBox has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.h [hunk 1]
```diff
@@ -319,6 +319,17 @@
...
 
+/**
+ * Clean up background session ID from IDs reconnected from extension
+ *
+ * @param backgroundSessionId   Id to clean
+ * @param error error if fail to complete
+ *
+ * @return YES if successfully cleaned up, NO if failed
+ */
+- (BOOL)cleanUpExtensionBackgrounSessionIdIfExists:(NSString * _Nonnull)backgroundSessionId
```
Comment (jlawton) [inline BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.h]: Maybe just `- (BOOL)cleanUpExtensionBackgroundSessionID:error:`?

File: BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m [hunk 2]
```diff
@@ -566,19 +569,55 @@ - (BOOL)cleanUpBackgroundSessionIfPossibleGivenUserID:(NSString * _Nonnull)userI
     if (backgroundSessionID != nil && [backgroundSessionID isEqualToString:[self backgroundSessionIdentifier]] == NO) {
         //this background session is not self.backgroundSession, most likely we are running inside the main app
         //and this background session is from an extension, clean it up if it has no more pending tasks
+
```
Comment (jlawton) [inline BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m]: This method strikes me as a good candidate for early returns. Several nested if statements with reversed cleanup is harder to read, imo.

File: BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.h (hunk 1 above)
Comment (jlawton) [reply BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.h]: Either way, there's a typo.

**Comment by jlawton** [comment]
Could we add a brief description in the log about the problem we're fixing and why we expect this to be a fix?

File: BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.h (hunk 1 above)
Comment (thuyBox) [reply BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.h]: done

File: BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m (hunk 2 above)
Comment (thuyBox) [reply BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m]: done

**Review by jlawton — APPROVED** [review]

**Pull request merged**